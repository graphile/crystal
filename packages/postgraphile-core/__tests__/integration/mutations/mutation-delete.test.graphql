mutation {
  a: deletePost(input: { nodeId: "WyJwb3N0cyIsMV0=" }) { ...deletePostPayload }
  b: deletePost(input: { nodeId: "WyJwb3N0cyIsMl0=", clientMutationId: "hello" }) { ...deletePostPayload }
  c: deletePost(input: { nodeId: "WyJwb3N0cyIsMjAwMF0=" }) { ...deletePostPayload }
  d: deletePost(input: { nodeId: "WyJwb3N0cyIsM10=", clientMutationId: "world" }) { ...deletePostPayload }
  d2: deleteTypeById(input: { id: 11, clientMutationId: "throw error" }) { clientMutationId deletedTypeId }
  e: deletePostById(input: { id: 6 }) { ...deletePostPayload }
  f: deletePostById(input: { id: 9, clientMutationId: "hello" }) { ...deletePostPayload }
  g: deletePostById(input: { id: 2000 }) { ...deletePostPayload }
  h: deletePostById(input: { id: 11, clientMutationId: "world" }) { ...deletePostPayload }
  i: deleteCompoundKey(input: { nodeId: "WyJjb21wb3VuZF9rZXlzIiw0LDNd" }) { ...deleteCompoundKeyPayload }
  j: deleteCompoundKeyByPersonId1AndPersonId2(input: { personId1: 2, personId2:3 }) { ...deleteCompoundKeyPayload }
	k: deletePersonByEmail(input: { email: "budd.deey@email.com" }) { ...deletePersonPayload }
	l: deletePersonByEmail(input: { email: "graphile-build.issue.27.exists@example.com" }) { ...deletePersonPayload2 }
}

fragment deletePostPayload on DeletePostPayload {
  clientMutationId
  deletedPostId
  post {
    nodeId
    id
    headline
    authorId
  }
  query { nodeId }
}

fragment deleteCompoundKeyPayload on DeleteCompoundKeyPayload {
  clientMutationId
  deletedCompoundKeyId
  compoundKey {
    nodeId
    personId1
    personId2
    personByPersonId1 {
      nodeId
      name
    }
    personByPersonId2 {
      nodeId
      name
    }
  }
  query { nodeId }
}

fragment deletePersonPayload on DeletePersonPayload {
  clientMutationId
  deletedPersonId
  query { nodeId }
}
fragment deletePersonPayload2 on DeletePersonPayload {
  clientMutationId
  deletedPersonId
  person {
    nodeId
    id
    name
    email
    issue27UserExists: exists(email: "graphile-build.issue.27.exists@example.com")
  }
  query { nodeId }
}
