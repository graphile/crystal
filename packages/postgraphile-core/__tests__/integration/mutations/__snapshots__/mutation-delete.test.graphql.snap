// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute 1`] = `
Object {
  "data": Object {
    "a": Object {
      "clientMutationId": null,
      "deletedPostId": "WyJwb3N0cyIsMV0=",
      "post": Object {
        "authorId": 2,
        "headline": "No… It’s a thing; it’s like a plan, but with more greatness.",
        "id": 1,
        "nodeId": "WyJwb3N0cyIsMV0=",
      },
      "query": Object {
        "nodeId": "query",
      },
    },
    "b": Object {
      "clientMutationId": "hello",
      "deletedPostId": "WyJwb3N0cyIsMl0=",
      "post": Object {
        "authorId": 1,
        "headline": "I hate yogurt. It’s just stuff with bits in.",
        "id": 2,
        "nodeId": "WyJwb3N0cyIsMl0=",
      },
      "query": Object {
        "nodeId": "query",
      },
    },
    "c": null,
    "d": Object {
      "clientMutationId": "world",
      "deletedPostId": "WyJwb3N0cyIsM10=",
      "post": Object {
        "authorId": 1,
        "headline": "Is that a cooking show?",
        "id": 3,
        "nodeId": "WyJwb3N0cyIsM10=",
      },
      "query": Object {
        "nodeId": "query",
      },
    },
    "d2": null,
    "e": Object {
      "clientMutationId": null,
      "deletedPostId": "WyJwb3N0cyIsNl0=",
      "post": Object {
        "authorId": 3,
        "headline": "Stop talking, brain thinking. Hush.",
        "id": 6,
        "nodeId": "WyJwb3N0cyIsNl0=",
      },
      "query": Object {
        "nodeId": "query",
      },
    },
    "f": Object {
      "clientMutationId": "hello",
      "deletedPostId": "WyJwb3N0cyIsOV0=",
      "post": Object {
        "authorId": 3,
        "headline": "You know how I sometimes have really brilliant ideas?",
        "id": 9,
        "nodeId": "WyJwb3N0cyIsOV0=",
      },
      "query": Object {
        "nodeId": "query",
      },
    },
    "g": null,
    "h": Object {
      "clientMutationId": "world",
      "deletedPostId": "WyJwb3N0cyIsMTFd",
      "post": Object {
        "authorId": 3,
        "headline": "They’re not aliens, they’re Earth…liens!",
        "id": 11,
        "nodeId": "WyJwb3N0cyIsMTFd",
      },
      "query": Object {
        "nodeId": "query",
      },
    },
    "i": Object {
      "clientMutationId": null,
      "compoundKey": Object {
        "nodeId": "WyJjb21wb3VuZF9rZXlzIiw0LDNd",
        "personByPersonId1": Object {
          "name": "Kathryn Ramirez",
          "nodeId": "WyJwZW9wbGUiLDRd",
        },
        "personByPersonId2": Object {
          "name": "Budd Deey",
          "nodeId": "WyJwZW9wbGUiLDNd",
        },
        "personId1": 4,
        "personId2": 3,
      },
      "deletedCompoundKeyId": "WyJjb21wb3VuZF9rZXlzIiw0LDNd",
      "query": Object {
        "nodeId": "query",
      },
    },
    "j": Object {
      "clientMutationId": null,
      "compoundKey": Object {
        "nodeId": "WyJjb21wb3VuZF9rZXlzIiwyLDNd",
        "personByPersonId1": Object {
          "name": "Sara Smith",
          "nodeId": "WyJwZW9wbGUiLDJd",
        },
        "personByPersonId2": Object {
          "name": "Budd Deey",
          "nodeId": "WyJwZW9wbGUiLDNd",
        },
        "personId1": 2,
        "personId2": 3,
      },
      "deletedCompoundKeyId": "WyJjb21wb3VuZF9rZXlzIiwyLDNd",
      "query": Object {
        "nodeId": "query",
      },
    },
    "k": Object {
      "clientMutationId": null,
      "deletedPersonId": "WyJwZW9wbGUiLDNd",
      "query": Object {
        "nodeId": "query",
      },
    },
    "l": Object {
      "clientMutationId": null,
      "deletedPersonId": "WyJwZW9wbGUiLDZd",
      "person": Object {
        "email": "graphile-build.issue.27.exists@example.com",
        "id": 6,
        "issue27UserExists": false,
        "name": "Twenty Seventwo",
        "nodeId": "WyJwZW9wbGUiLDZd",
      },
      "query": Object {
        "nodeId": "query",
      },
    },
  },
  "errors": Array [
    [GraphQLError: No values were deleted in collection 'posts' because no values you can delete were found matching these criteria.],
    [GraphQLError: Nope.],
    [GraphQLError: No values were deleted in collection 'posts' because no values you can delete were found matching these criteria.],
  ],
}
`;
